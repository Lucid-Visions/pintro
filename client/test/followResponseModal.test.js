import React from "react";
import { create, act } from "react-test-renderer";
import FollowModal from "../src/components/followRequestModal";

describe("<FollowRequestModal />", () => {
  let props;
  let accepted;

  const connection = {
    startTime: 15866159730000,
    intent: "Unit Testing",
    message: "I want to help you do unit tests",
  };

  const createProps = (isVisible) => ({
    visible: isVisible,
    toggle: jest.fn(() => props = createProps(!props.visible)),
    onModalHide: jest.fn(() => props = createProps(!props.visible)),
    accepted: false,
    data: {
      status: "PENDING",
      follower: {
        _id: "431",
        name: "Kim",
        profile_picture: null,
        bio: "Kim is a very good app tester",
      },
    },
    connection,
    respond: jest.fn((x) => (accepted = x)),
    profile_picture: null,
    connectModal: null,
  });

  beforeEach(() => {
      props = createProps(true)
  })

  test("snapshot of the component", () => {
    const component = create(<FollowModal {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('test backdrop to close the modal', () => {
    const component = create(<FollowModal {...props} />)
    const instance = component.root
    const modal = instance.findByType("Modal")
    const backdrop = modal.children[0].props

    expect(props.visible).toBe(true);
    // click on backdrop
    backdrop.onPress();
    expect(props.visible).toBe(false);
  })

  test('test close modal button', () => {
    const component = create(<FollowModal {...props} />)
    const instance = component.root
    const modal = instance.findByType("Modal")
    const closeBtn = modal.props

    expect(props.visible).toBe(true);
    // click on backdrop
    closeBtn.onModalHide();
    expect(props.visible).toBe(false);
  })
});
