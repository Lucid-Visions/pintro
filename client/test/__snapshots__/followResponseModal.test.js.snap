// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowRequestModal /> snapshot of the component 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FCFCFC",
        "borderRadius": 20,
        "flex": 1,
        "height": 333.5,
        "justifyContent": "space-evenly",
        "margin": 37.5,
        "marginTop": 296.44444444444446,
        "maxHeight": 350,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "width": 681.8181818181818,
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 625,
          }
        }
      >
        <Image
          style={
            Object {
              "alignSelf": "center",
              "borderRadius": 30,
              "height": 60,
              "overflow": "hidden",
              "width": 60,
            }
          }
        />
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "poppins-bold",
                "fontSize": 13,
                "margin": "auto",
                "textAlign": "left",
              }
            }
          >
            Kim
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "poppins-medium",
                "fontSize": 13,
                "textAlign": "left",
                "width": 200,
              }
            }
          >
            Kim is a very good app tester...
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 13,
                  "marginRight": 5,
                  "resizeMode": "contain",
                  "width": 10,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "black",
                  "fontFamily": "poppins-light",
                  "fontSize": 11,
                  "margin": "auto",
                  "textAlign": "left",
                }
              }
            >
               
              in NaN years
               
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 13,
                "marginTop": 4,
                "width": 13,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "#F8AE39",
              }
            }
          >
             
             
            km
             
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "backgroundColor": "#DDDDDD",
            "borderRadius": 20,
            "flexDirection": "column",
            "justifyContent": "space-between",
            "padding": 15,
            "width": 625,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "poppins-bold",
              "fontSize": 13,
              "margin": "auto",
              "textAlign": "left",
            }
          }
        >
          Intent:
        </Text>
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "poppins-light",
              "fontSize": 13,
              "margin": "auto",
              "textAlign": "left",
            }
          }
        >
          Unit Testing
        </Text>
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "poppins-bold",
              "fontSize": 13,
              "margin": "auto",
              "textAlign": "left",
            }
          }
        >
          Message:
        </Text>
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "poppins-light",
              "fontSize": 13,
              "margin": "auto",
              "textAlign": "left",
            }
          }
        >
          I want to help you do unit tests
        </Text>
      </View>
      <View
        flexDirection="row"
        style={
          Object {
            "alignItems": "space-between",
            "borderColor": "orange",
            "borderWidth": 0,
            "justifyContent": "space-between",
            "marginTop": 20,
            "width": 625,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 7.5,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": "47%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "poppins-medium",
                "fontSize": 10,
              }
            }
          >
            ACCEPT
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "black",
              "borderRadius": 7.5,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 35,
              "justifyContent": "center",
              "opacity": 1,
              "width": "47%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "poppins-medium",
                "fontSize": 10,
              }
            }
          >
            DECLINE
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
